client/src/components/Navbar.jsx

```diff
@@ -86,12 +86,10 @@ const Navbar = (props) => {
 										
 										<div className="m-auto w-full">
 											<DangerButton
-												link="/"
+												link="/api/logout"
 												target="_self"
 												text="Sign Out"
 												onClick={(e) => {
-													e.preventDefault();
-
 													localStorage.removeItem("token");
```

Comment:
Saving `token` into local storage is convenient, but also vulnerable to XSS exploits. If the hacker manages to inject a script on your website, they can easily access & steal your `token`. Instead, you should make `token` a **httpOnly** cookie. For logout you can make an API that simply deletes the cookie.